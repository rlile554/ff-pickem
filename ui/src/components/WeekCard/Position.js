import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { FiLock } from 'react-icons/fi';

const Container = styled.div`
    width: 100%;
    height: 50px;
    border-top: 1px solid #979797;
    display: flex;
    align-items: center;
    padding-left: 2px;
`;

const PosDiv = styled.div`
    height: 40px;
    width: 40px;
    border-radius: 50%;
    background-color: #5b5b5b;
    display: flex;
    align-items: center;
    justify-content: center;
    color: white;
    font-weight: bold;
`;

const IconDiv = styled.div`
    width: 25px;
    height: 25px;
    font-size: 20px;
    padding-top: 5px;
`;

const Name = styled.div`
    font-weight: bold;
    width: 180px;
    text-align: center;
`;

const Points = styled.div`
    font-weight: bold;
    text-align: right;
    width: 60px;
`;

const Position = props => {
    const { position, locked, name, points } = props;
    return (
        <Container>
            <PosDiv id="position">{position}</PosDiv>
            <IconDiv id="iconDiv">{locked && <FiLock />}</IconDiv>
            <Name id="name">{name}</Name>
            <Points id="points">{points}</Points>
        </Container>
    );
};

Position.propTypes = {
    //The position of the player
    position: PropTypes.string,
    //wether or not the pick is locked in
    locked: PropTypes.bool,
    //the name of the player
    name: PropTypes.string,
    //the numer of ponts the player currently has
    points: PropTypes.number
};

export default Position;
