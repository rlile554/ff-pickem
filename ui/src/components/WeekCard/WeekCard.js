import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FiEdit } from 'react-icons/fi';

import Position from './Position';


const Container = styled.div`
    width: 320px;
    height: 340px;
    border: solid 1px #979797;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
`;

const Week = styled.div`
    text-transform: uppercase;
    font-size: 48px;
    font-weight: bold;
`;
const Score = styled.div`
    font-size: 24px;
    fontweight: bold;
`;

const IconDiv = styled.div`
    margin-top: 4px;
    margin-right: 4px;
    align-self: flex-end;
    font-size: 20px;
`;

const PositionContainer = styled.div`
    width: 100%;
`;

const getPointsText = (points) => {
    return points? `points: ${points}` : 'points: 0';
}

const WeekCard = props => {
    const { weekNumber, points } = props;
    return (
        <Container>
            <IconDiv>
                <FiEdit />
            </IconDiv>
            <Week>Week {weekNumber}</Week>
            <Score>{getPointsText(points)}</Score>
            <PositionContainer id="positionsContainer">
                <Position position={'QB'} locked name={"Tom Brady"} points={12345} />
                <Position position={'RB'} name={"Mark Ingram"} points={2233} />
                <Position position={'WR'} name={"Sammie Watkins"} points={323} />
                <Position position={'D'} locked name={"Patriots"} points={1} />
            </PositionContainer>
        </Container>
    );
};

WeekCard.propTypes = {
    weekNumber: PropTypes.number.isRequired,
    points: PropTypes.number
}



export default WeekCard;
