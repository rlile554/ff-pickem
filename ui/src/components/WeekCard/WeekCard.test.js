import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import WeekCard from './WeekCard';

describe('WeekCard', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<WeekCard weekNumber={1}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches the snapshot', () => {
        const component = renderer.create(<WeekCard weekNumber={1} points={12345} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});