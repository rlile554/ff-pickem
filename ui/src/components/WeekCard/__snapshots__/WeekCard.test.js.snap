// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekCard matches the snapshot 1`] = `
<div
  className="sc-EHOje UBhwR"
>
  <div
    className="sc-htoDjs bPOjW"
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
      />
      <path
        d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
      />
    </svg>
  </div>
  <div
    className="sc-bZQynM hElvZM"
  >
    Week 
    1
  </div>
  <div
    className="sc-gzVnrw ddarIT"
  >
    points: 12345
  </div>
  <div
    className="sc-dnqmqq hMmkfQ"
    id="positionsContainer"
  >
    <div
      className="sc-bdVaJa hvvqAj"
    >
      <div
        className="sc-bwzfXH fzysLj"
        id="position"
      >
        QB
      </div>
      <div
        className="sc-htpNat klVHJR"
        id="iconDiv"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="11"
            rx="2"
            ry="2"
            width="18"
            x="3"
            y="11"
          />
          <path
            d="M7 11V7a5 5 0 0 1 10 0v4"
          />
        </svg>
      </div>
      <div
        className="sc-bxivhb bfViTE"
        id="name"
      >
        Tom Brady
      </div>
      <div
        className="sc-ifAKCX jlmnUT"
        id="points"
      >
        12345
      </div>
    </div>
    <div
      className="sc-bdVaJa hvvqAj"
    >
      <div
        className="sc-bwzfXH fzysLj"
        id="position"
      >
        RB
      </div>
      <div
        className="sc-htpNat klVHJR"
        id="iconDiv"
      />
      <div
        className="sc-bxivhb bfViTE"
        id="name"
      >
        Mark Ingram
      </div>
      <div
        className="sc-ifAKCX jlmnUT"
        id="points"
      >
        2233
      </div>
    </div>
    <div
      className="sc-bdVaJa hvvqAj"
    >
      <div
        className="sc-bwzfXH fzysLj"
        id="position"
      >
        WR
      </div>
      <div
        className="sc-htpNat klVHJR"
        id="iconDiv"
      />
      <div
        className="sc-bxivhb bfViTE"
        id="name"
      >
        Sammie Watkins
      </div>
      <div
        className="sc-ifAKCX jlmnUT"
        id="points"
      >
        323
      </div>
    </div>
    <div
      className="sc-bdVaJa hvvqAj"
    >
      <div
        className="sc-bwzfXH fzysLj"
        id="position"
      >
        D
      </div>
      <div
        className="sc-htpNat klVHJR"
        id="iconDiv"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="11"
            rx="2"
            ry="2"
            width="18"
            x="3"
            y="11"
          />
          <path
            d="M7 11V7a5 5 0 0 1 10 0v4"
          />
        </svg>
      </div>
      <div
        className="sc-bxivhb bfViTE"
        id="name"
      >
        Patriots
      </div>
      <div
        className="sc-ifAKCX jlmnUT"
        id="points"
      >
        1
      </div>
    </div>
  </div>
</div>
`;
